sstables/generation_type.hh

```diff
@@ -57,9 +64,31 @@ public:
             ec == std::errc() && ptr == s.data() + s.size()) {
             return generation_type(int_value);
         } else {
-            throw std::invalid_argument(fmt::format("invalid UUID: {}", s));
+            static const boost::regex pattern("([0-9a-z]{4})_([0-9a-z]{4})_([0-9a-z]{5})([0-9a-z]{13})");
+            boost::smatch match;
+            if (!boost::regex_match(s, match, pattern)) {
+                throw std::invalid_argument(fmt::format("invalid UUID: {}", s));
+            }
+            utils::UUID_gen::decimicroseconds timestamp = {};
+            auto decode_base36 = [](const std::string& s) {
+                std::size_t pos{};
+                auto n = std::stoull(s, &pos, 36);
+                if (pos != s.size()) {
+                    throw std::invalid_argument(fmt::format("invalid part in UUID: {}", s));
+                }
+                return n;
+            };
+            timestamp += std::chrono::days{decode_base36(match[1])};
+            timestamp += std::chrono::seconds{decode_base36(match[2])};
+            timestamp += ::utils::UUID_gen::decimicroseconds{decode_base36(match[3])};
```

Comment:
We must have an UUID string parser already, we have an UUID type in CQL, which parses values from string.

Reply:
i want to be compatible with cassandra. the existing parser does not use base36, while the cassandra does. see https://github.com/apache/cassandra/blob/edd35badaf6e25726b0e46a4cddf9cfe42f0c11a/src/java/org/apache/cassandra/io/sstable/UUIDBasedSSTableId.java#L4